Zend/zend_hash.c

```diff
@@ -1331,6 +1331,17 @@ ZEND_API void ZEND_FASTCALL zend_hash_rehash(HashTable *ht)
 		if (!(HT_FLAGS(ht) & HASH_FLAG_UNINITIALIZED)) {
 			ht->nNumUsed = 0;
 			HT_HASH_RESET(ht);
+			/* Even if the array is empty, we still need to reset the iterator positions. */
```

Comment:
This might also have to reset `nInternalPointer`, let me see if I can trigger that.